packages/helpers/src/hooks/useMountedRef.js

```diff
@@ -0,0 +1,14 @@
+import React from "react";
+
+export default function useMountedRef() {
+  const mountedRef = React.useRef(false);
+
+  React.useEffect(() => {
+    mountedRef.current = true;
+    return function cleanup() {
+      mountedRef.current = false;
+    };
+  }, []);
+
+  return mountedRef;
```

Comment:
Late to the game, but why not return `mountedRef.current` here? The end user probably never cares about the ref itself, they're interested in the boolean value of whether we're mounted or not (and so `ref` in the hook name looks irrelevant to me)

Reply:
@mikrotron @alexzherdev 

Ref is safer to use with effects. Let's say you want to make a request `on first mount`: 

```js
useEffect(() => {
  fetch().then(() => {
    if (ref.current) { // do something }
  })
}, [ref]);
```

Since `ref` is immutable, the change will not trigger an effect to reapply. 

But if we do `const isMounted = useMounted()`

then we have to pass it as parameter. And the effect will be reapplied when the component dismounts.  

```js
useEffect(() => {
  fetch().then(() => {
    if (isMounted) { // do something }
  })
}, [isMounted]);
```

Reply:
Hmm I see. In the former case, does eslint warn if you don't specify `ref` as dependency? I think refs are guaranteed to keep referential integrity.
I'm assuming in the latter case, if you don't specify `isMounted`, it should definitely warn.

Reply:
@alexzherdev Yes, it'll warn you. 
But with `ref` it is safe to add it as a dependency because of `referential integrity`. 

We might think of having a warning instead of an error (as we have now) for that. But I'm worried that people might ignore warnings 🤔